src/Corax/Queries/UnaryMatch.Ordinal.cs

```diff
@@ -133,6 +135,122 @@ private static int FillFuncSequenceAny<TComparer>(ref UnaryMatch<TInner, TValueT
 
             return storeIdx;
         }
+        
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        private static int BinarySearch(TermQueryItem[] item, ReadOnlySpan<byte> value)
+        {
+            int l = 0;
+            int r = item.Length - 1;
+            while (l <= r)
+            {
+                var pivot = (l + r) >> 1;
+                switch (item[pivot].Item.Span.SequenceCompareTo(value))
+                {
+                    case 0:
+                        return pivot;
+                    case < 0:
+                        l = pivot + 1;
+                        break;
+                    default:
+                        r = pivot - 1;
+                        break;
+                }
+            }
+
+            return -1;
+        }
+
+        [SkipLocalsInit]
+        private static int FillFuncSequenceAllIn(ref UnaryMatch<TInner, TValueType> match, Span<long> matches)
+        {
+            var value = ((TermQueryItem[])(object)match._value);
+            using var bitset = new BitSet(value.Length);
```

Comment:
The primitive to do this already exist at `Sparrow.Binary.PtrBitVector` . In this case, this causes a gen 1 allocation. Instead of this, construct a stackalloc  temporary value. That could conflict though with the `SkipLocalsInit` attribute though, because we want it to be initialized with 0s.

Reply:
fixed